test/units/utils/test_encrypt.py

```diff
@@ -249,3 +249,55 @@ def test_passlib_bcrypt_salt(recwarn):
 
     result = passlib_obj.hash(secret, salt=repaired_salt, ident=ident)
     assert result == expected
+
+
+def test_do_encrypt_no_lib(mocker):
```

Comment:
I think we should start typing the tests when adding new ones (existing tests would need a substantial effort, OTOH, so that's a different story).

Pytest and most of its prominent fixture-providing plugins expose public API for typing so there's nothing stopping us from starting to leverage it. Here's how you can do it.

```python
import pytest_mock as _ptm
```

```suggestion
def test_do_encrypt_no_lib(mocker: _ptm.MockerFixture) -> None:
```